### ISSUE TITLE
Feature Request: 'arange_like/enumerate' operator

### ISSUE THREAD

**Issue by fhieber** [apache/incubator-mxnet#14574]
It would be quite useful to have an operator that can create ranges based on an existing 'shape_like' symbol, similar to the ops `zeros_like`, `ones_like`, and other `*_like`.

For example, I would like to enumerate the positions of a symbol with dynamic shape:
```python
x = mx.sym.Variable('x')
r = mx.sym.arange_like(x, start=0)
r.eval(x=mx.nd.ones((4,))) # should give: [0, 1, 2, 3]
r.eval(x=mx.nd.ones((2,))) # should give: [0, 1]
r2 = mx.sym.arange_like(x, start=5)
r.eval(x=mx.nd.ones((2, 2))) # should give: [[5, 6], [7, 8]]
```

Another name could be 'enumerate'.

**Comment by vdantu** [comment]
@mxnet-label-bot add [operator, feature]

**Comment by fhieber** [comment]
Resolved by #15400
